Wrappers/Python/cil/framework/framework.py

```diff
@@ -1795,25 +1808,70 @@ class DataContainer(object):
     '''Generic class to hold data
     
     Data is currently held in a numpy arrays'''
-    
+
+    @property
+    def geometry(self):
+        return None
+
+    @geometry.setter
+    def geometry(self, val):
+        if val is not None:
+            raise TypeError("DataContainers cannot hold a geometry, use ImageData or AcquisitionData instead")
+
+    @property
+    def dimension_labels(self):
+        default_labels = [0]*self.number_of_dimensions
```

Comment:
maybe this could go in the if None statement as we don't need to create the labels all the time if we have one set already.